src/kernel/sporadic.c

```diff
@@ -199,34 +221,12 @@ void refill_update(sched_context_t *sc, ticks_t new_period, ticks_t new_budget,
         refill_t new = { .rAmount = (new_budget - REFILL_HEAD(sc).rAmount),
                          .rTime = REFILL_HEAD(sc).rTime + new_period
                        };
-        refill_add_tail(sc, new);
+        schedule_used(sc, new);
```

Comment:
what is the differefnce between schedule_used and refill_add_tail? I recall them being actually semantically different.

Reply:
`refill_add_tail` doesn't do any checking, it just adds the tail. `schedule_used` adds a tail but does some checking and moving of refills. It seemingly serves to satisfy some properties that I don't think I could find in the original paper.

Reply:
Right. In that case it was written to reduce branches and avoid checks we know we don't need.

Reply:
If this isn't needed it would be better to avoid merging this one.